lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -58,6 +58,21 @@ class TensorEmptyConversionPattern
   matchAndRewrite(tensor::EmptyOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
 
+    // BroadcastOp will be folded away so its EmptyOp should be erased.
+    bool allBroadcastOpConsumers = true;
+    if (not op->getUsers().empty()) {
```

Comment:
Can it happen that broadcast removal is done first and the list of users of this op is empty? 
In that case should we remove this empty op?

Reply:
I think that in conversion pass nothing is removed until pass is completed, just internally flagged for removal.

Reply:
Yes @nobradovictt is right, if users list was empty, something else must have caused it.